aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -1200,6 +1190,24 @@ Tensor NestedTensor_from_buffer_copy(
       contiguous_buffer.clone(), nt_size, nt_strides, nt_offsets);
 }
 
+Tensor _from_buffer_maybe_view(
+    const Tensor& buffer,
+    const Tensor& reference_tensor) {
+  TORCH_CHECK(
+      !buffer.is_nested(),
+      "Can only a create Nested Tensor from a normal tensor buffer");
+  TORCH_CHECK(buffer.dim() == 1, "The input buffer must be flat");
+  // Because of broadcasting ops we need to call contiguous on the buffer
+  auto contiguous_buffer = buffer.contiguous();
+  // example is NT -> to_buffer().sum().backward()
+  // Problem: if contigious is actually needed
+  // AT_ASSERT(self__storage_saved.value().is_alias_of(result.storage()));
+  // This will fail because we are creating a new storage.
+  // I am not sure how we get around this without make the another more native
+  // function And this is like reshape which is maybe aliasing. this sucks lol
```

Comment:
That is what this _from_buffer_maybe_view function is doing. It is a composite implicit op that returns a view when the buffer is contiguous otherwise calls contiguous than returns a view. Wasn't as annoying as I made it sound in the comment here.